File: js/src/trees/binaryTreeTraverser.js
function BinaryTreeTraverser() { }

BinaryTreeTraverser.prototype = (function() {
  function preOrder(node) {
    return preOrderHelper(node).trim();
  }

  function preOrderHelper(node) {
    var output = '';

    output += node.value + ' ';

    if(node.left) {
      output += preOrderHelper(node.left);
    }

    if(node.right) {
      output += preOrderHelper(node.right);
    }

    return output;
  }

  function inOrder(node) {
    return inOrderHelper(node).trim();
  }

  function inOrderHelper(node) {
    var output = '';

    if(node.left) {
      output += inOrderHelper(node.left);
    }

    output += node.value + ' ';

    if(node.right) {
      output += inOrderHelper(node.right);
    }

    return output;
  }

  function postOrder(node) {
    return postOrderHelper(node).trim();
  }

  function postOrderHelper(node) {
    var output = '';

    if(node.left) {
      output += postOrderHelper(node.left);
    }

    if(node.right) {
      output += postOrderHelper(node.right);
    }

    output += node.value + ' ';

    return output;
  }

  return {
    preOrder: preOrder,
    inOrder: inOrder,
    postOrder: postOrder
  }
}.call(this))

File: C#/DataStructures/DataStructures/trees/BinaryTreeTraverser.cs
namespace DataStructures.trees
{
	public class BinaryTreeTraverser
	{
		public string PreOrder<T>(IBinaryTree<T> tree)
		{
            return PreOrderHelper(tree).Trim();
		}

        private string PreOrderHelper<T>(IBinaryTree<T> tree)
        {
            string output = "";

            output += tree.Value + " ";

            if (tree.Left != null)
                output += PreOrderHelper(tree.Left);

            if (tree.Right != null)
                output += PreOrderHelper(tree.Right);

            return output;
        }

		public string InOrder<T>(IBinaryTree<T> tree)
		{
            return InOrderHelper(tree).Trim();
		}

        private string InOrderHelper<T>(IBinaryTree<T> tree)
        {
            string output = "";

            if (tree.Left != null)
                output += InOrderHelper(tree.Left);

            output += tree.Value + " ";

            if (tree.Right != null)
                output += InOrderHelper(tree.Right);

            return output;
        }

		public string PostOrder<T>(IBinaryTree<T> tree)
		{
            return PostOrderHelper(tree).Trim();
		}

        public string PostOrderHelper<T>(IBinaryTree<T> tree)
        {
            string output = "";

            if (tree.Left != null)
                output += PostOrderHelper(tree.Left);

            if (tree.Right != null)
                output += PostOrderHelper(tree.Right);

            output += tree.Value + " ";

            return output;
        }
    }
}
